app/src/composables/tips.ts

```diff
@@ -0,0 +1,68 @@
+import { TIPS_ADDRESS } from '@/constant'
+import { ethers } from 'ethers'
+import ABI from '../abi/tips.json'
+import { onMounted, ref } from 'vue'
+import { useToast } from './toast'
+import { ToastType } from '@/types'
+import type { AddressLike } from 'ethers'
+
+export function useTips() {
+  const { showToast } = useToast()
+  const provider = new ethers.BrowserProvider((window as any).ethereum)
+
+  const contract = new ethers.Contract(TIPS_ADDRESS, ABI, provider)
+  const pushTipLoading = ref(false)
+  const sendTipLoading = ref(false)
+
+  async function pushTip(addresses: AddressLike[], amount: number) {
+    pushTipLoading.value = true
+
+    try {
+      const tx = await contract.pushTip(addresses, {
+        value: ethers.parseEther(amount.toString())
+      })
+      await tx.wait()
+
+      showToast(ToastType.Success, 'Tip pushed successfully')
```

Comment:
Separation of concern

Your composable Tips don't need to use Toast. 

For example if in the future you don't need to use toast every times you push tip but only when you push tips on a specific page.
How you will handle this ?
So you need to avoid to depend on it. But rather you can simply return the result and the component will decide what to do with this information.

Reply:
> Separation of concern
> 
> Your composable Tips don't need to use Toast.
> 
> For example if in the future you don't need to use toast every times you push tip but only when you push tips on a specific page. How you will handle this ? So you need to avoid to depend on it. But rather you can simply return the result and the component will decide what to do with this information.

Thank you for pointing this out, I've seen it used inside of API calls and a lot of places. If this functionality changes a lot of code will need to be updated/changed without any actual productivity, just maintenance